definitions/npm/yargs_v16.x.x/flow_v0.104.x-/yargs_v16.x.x.js

```diff
@@ -0,0 +1,296 @@
+    alias: string | Array<string>,
+    array: boolean,
+    boolean: boolean,
+    choices: Array<mixed>,
+    coerce: (arg: {[key: string]: any, ...} | any) => mixed,
+    config: boolean,
+    configParser: (configPath: string) => { [key: string]: mixed, ... },
+    conflicts: string | Array<string> | { [key: string]: string, ... },
+    count: boolean,
+    default: mixed,
+    defaultDescription: string,
+    demandOption: boolean | string,
+    desc: string,
+    describe: string,
+    description: string,
+    global: boolean,
+    group: string,
+    implies: string | { [key: string]: string, ... },
+    nargs: number,
+    normalize: boolean,
+    number: boolean,
+    required: boolean,
+    requiresArg: boolean,
+    skipValidation: boolean,
+    string: boolean,
+    type: "array" | "boolean" | "count" | "number" | "string",
+    ...
+  }>;
+
+  declare type CommonModuleObject = {|
+    command?: string | Array<string>,
+    aliases?: Array<string> | string,
+    builder?: { [key: string]: Options, ... } | ((yargsInstance: Yargs) => mixed),
+    handler?: ((argv: Argv) => void) | ((argv: Argv) => Promise<void>)
+  |};
+
+  declare type ModuleObjectDesc = {|
+    ...CommonModuleObject,
+    desc?: string | false
+  |};
+
+  declare type ModuleObjectDescribe = {|
+    ...CommonModuleObject,
+    describe?: string | false
+  |};
+
+  declare type ModuleObjectDescription = {|
+    ...CommonModuleObject,
+    description?: string | false
+  |};
+
+  declare type ModuleObject =
+    | ModuleObjectDesc
+    | ModuleObjectDescribe
+    | ModuleObjectDescription;
+
+  declare type MiddleWareCallback =
+    | (argv: Argv, yargsInstance?: Yargs) => void
+    | (argv: Argv, yargsInstance?: Yargs) => Promise<void>;
+
+  declare type Middleware = MiddleWareCallback | Array<MiddleWareCallback>;
+
+  declare class Yargs {
+    (args: Array<string>): Yargs;
+
+    alias(key: string, alias: string): this;
+    alias(alias: { [key: string]: string | Array<string>, ... }): this;
+    argv: Argv;
+    array(key: string | Array<string>): this;
+    boolean(parameter: string | Array<string>): this;
+    check(fn: (argv: Argv, options: Array<string>) => ?boolean): this;
+    choices(key: string, allowed: Array<string>): this;
+    choices(allowed: { [key: string]: Array<string>, ... }): this;
+    coerce(key: string, fn: (value: any) => mixed): this;
+    coerce(object: { [key: string]: (value: any) => mixed, ... }): this;
+    coerce(keys: Array<string>, fn: (value: any) => mixed): this;
+
+    command(
+      cmd: string | Array<string>,
+      desc: string | false,
+      builder?: { [key: string]: Options, ... } | ((yargsInstance: Yargs) => mixed),
+      handler?: Function
+    ): this;
+
+    command(
+      cmd: string | Array<string>,
+      desc: string | false,
+      module: ModuleObject
+    ): this;
+
+    command(module: ModuleObject): this;
+
+    commandDir(
+      directory: string,
+      options?: {
+        exclude?: string | Function,
```

Comment:
`Function` -> `any`, if you could type it a bit more precisely?

Reply:
TBH I just copied the libdef from v15 ahah. I don't know this lib at all.
Do you really want me to look at this before landing? Obviously I didn't have the time in 2 weeks, so I'm not sure I'll be able to in the future either.

Reply:
He just means don't use `Function` because it alias' to `any`. I got the same comment here :)

Reply:
Yeah I understand the comment. What I'm saying is that I just copied this over from v15. So this isn't a regression from what we had before, and still providing some type fo the users of yargs v16 even if not perfect.

Reply:
Okay, not forcing you to update those ;)   
But now we need a rebase... sorry @julienw.